src/shmem_synchronization.h

```diff
@@ -30,6 +30,7 @@ shmem_internal_quiet(shmem_ctx_t ctx)
     if (0 != ret) { RAISE_ERROR(ret); }
 
     shmem_internal_membar();
+    shmem_transport_syncmem();
```

Comment:
Where possible, comments capturing rationale may be helpful.  In this case the syncmem is needed to make visible (i.e. to subsequent coherent load operations performed either by the local PE or another PE on the same node via the `shmem_ptr` API) the result of atomics that targeted the local process and were performed through the transport layer API.

Reply:
I think the rationale is the same for set/test lock, except that it should also provide visibility of operations performed within a previous critical section.

Reply:
Thanks @jdinan. Can you please review the comments I have added based on your suggestions? Let me know if these look ok.